src/util/whitespace/AstStringOptimizeStrategy.ts

```diff
@@ -356,42 +370,79 @@ export class AstStringOptimizeStrategy implements StringOptimizeStrategy {
 		whitespaceToAdd: string,
 		direction: PROCESS_DIRECTION
 	) {
-		let lastDeletedIndex = iIndex;
-		if (direction === PROCESS_DIRECTION.PRECEDING) {
-			for (let i = whitespaceToRemove.length - 1; i >= 0; i--) {
-				const sWhitespaceChar = whitespaceToRemove[i];
-				const iIndexToDelete =
-					iIndex - (whitespaceToRemove.length - 1 - i);
-				const sCharInMod = aOptimizedContent[iIndexToDelete];
-				if (sCharInMod === sWhitespaceChar) {
-					aOptimizedContent[iIndexToDelete] = "";
-					lastDeletedIndex = iIndexToDelete;
+		const aOptimizedContentBackup = aOptimizedContent.slice();
+
+		/**
+		 * collects actions to perform
```

Comment:
Maybe "Collects actions to perform a simulation of a modification and test it..." ???

This is hard to understand.. what is an action for? What is tested and why? An example could be useful.

Reply:
done